packages/jest-matchers/src/index.js

```diff
@@ -58,14 +65,37 @@ if (!global[GLOBAL_STATE]) {
 
 const expect: Expect = (actual: any): ExpectationObject => {
   const allMatchers = global[GLOBAL_STATE].matchers;
-  const expectation = {not: {}};
+  const expectation = {
+    not: {},
+    rejects: {not: {}},
+    resolves: {not: {}},
+  };
+
   Object.keys(allMatchers).forEach(name => {
     expectation[name] =
       makeThrowingMatcher(allMatchers[name], false, actual);
     expectation.not[name] =
       makeThrowingMatcher(allMatchers[name], true, actual);
   });
 
+  const resolveExpectation = {not: {}};
+  Object.keys(allMatchers).forEach(name => {
```

Comment:
it seems quite repetitive to receive all the keys every time. Can we avoid it and only iterate over them once?